### PR TITLE
feat(collection tags): frontend changes for tags in collection

### PR DESCRIPTION
**Pull request by seaerchin**
## Problem
We need to support tagging in our collections for the CLS use case. this PR adds the initial collection tags in 

## Solution
1. Add an internal `Tag` component for use
2. Add this `Tag` to the `CollectionCard`

### CODE REVIEW

File: packages/components/src/interfaces/internal/CollectionCard.ts [hunk 1]
```diff
@@ -2,14 +2,15 @@ import type { ImageProps } from "~/interfaces"
 import type { IsomerSiteProps, LinkComponentType } from "~/types"
 
 export interface Tag {
-  title: string
+  label: string
```
Comment (harishv7) [inline packages/components/src/interfaces/internal/CollectionCard.ts:5]: this is as discussed a categorical tag right? shouldn't it be having additional properties:

```
export interface Tag {
    category: string
    values: string[]
}
```

Comment (harishv7) [reply packages/components/src/interfaces/internal/CollectionCard.ts:5]: oops, saw the changes in the BE PR

**Review by harishv7 — APPROVED** [review]

File: packages/components/src/templates/next/components/internal/Tag/Tag.tsx [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+export const Tag = (
+  props: PropsWithChildren<
+    React.DetailedHTMLProps<
+      React.HTMLAttributes<HTMLParagraphElement>,
+      HTMLParagraphElement
+    >
+  >,
+) => {
+  return (
+    <div className="flex w-fit items-center justify-center gap-2 rounded-full bg-base-canvas-backdrop px-1.5 py-0.5 text-base-content-subtle">
```
Comment (adriangohjw) [inline packages/components/src/templates/next/components/internal/Tag/Tag.tsx:12]: question - why do we need to `flex` and `gap-2` these?

File: packages/components/src/templates/next/components/internal/CollectionCard/CollectionCard.tsx [hunk 3]
```diff
@@ -47,6 +49,13 @@ export const CollectionCard = ({
             <span title={itemTitle}>{itemTitle}</span>
           </Link>
         </h3>
+        {tags && tags.length > 0 && (
+          <div className="flex flex-wrap gap-2">
```
Comment (adriangohjw) [inline packages/components/src/templates/next/components/internal/CollectionCard/CollectionCard.tsx:53]: question - figma shows gap 6px which should be gap-1.5?

image

Comment (seaerchin) [reply packages/components/src/templates/next/components/internal/CollectionCard/CollectionCard.tsx]: good catch, updated!

File: packages/components/src/templates/next/components/internal/Tag/Tag.tsx (hunk 2 above)
Comment (seaerchin) [reply packages/components/src/templates/next/components/internal/Tag/Tag.tsx]: removed - double checked storybook to ensure no changes

**Comment by seaerchin** [comment]
* **#916** <a href="https://app.graphite.dev/github/pr/opengovsg/isomer/916?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#911** <a href="https://app.graphite.dev/github/pr/opengovsg/isomer/911?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/opengovsg/isomer/911?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

**Comment by seaerchin** [comment]
### Merge activity

* **Dec 19, 10:18 PM EST**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/opengovsg/isomer/911).

**Pull request merged**